lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -6026,6 +6027,79 @@ void ModuleEmitter::emitHWModule(HWModuleOp module) {
   currentModuleOp = nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// Emitter for files & file lists.
+//===----------------------------------------------------------------------===//
+
+class FileEmitter : public EmitterBase {
+public:
+  explicit FileEmitter(VerilogEmitterState &state) : EmitterBase(state) {}
+
+  void emit(emit::FileOp op);
+  void emit(emit::FileListOp op);
+
+private:
+  void emitOp(emit::VerbatimOp op);
+};
+
+void FileEmitter::emit(emit::FileOp op) {
+  for (Operation &op : *op.getBodyBlock()) {
+    TypeSwitch<Operation *, void>(&op)
+        .Case<emit::VerbatimOp>([&](auto op) { emitOp(op); })
+        .Default([&](auto op) {
+          op->emitError("cannot emit operation to file");
```

Comment:
Mega-nit: `emitOpError` may be better with a message that works with that format, i.e., `cannot be emitted to a file`.

This is only illegal at this point because more full support is not implemented correct? I.e., I'd expect this to work HW Modules. If this is intended to be very restrictive, then this smells like a missing verifier on `FileOp`.

Reply:
Yes, we'll have to pull in module/statement emitters here.